diesel_dynamic_schema/src/table.rs

```diff
@@ -112,3 +113,18 @@ impl<T, U> QueryId for Table<T, U> {
     type QueryId = ();
     const HAS_STATIC_QUERY_ID: bool = false;
 }
+
+impl<T, U> HasTable for Table<T, U>
+where
+    T: Default,
+    Self: QuerySource + AsQuery,
+{
+    type Table = Self;
+
+    fn table() -> Self {
+        Self {
+            name: T::default(),
```

Comment:
There is no way that this will work, as the delete statement needs to have access to the underlying table name.

It be possible to fix this by either looking at the existing `HasTable` impl for `SelectStatement` or at the `IntoUpdateTarget` impls for `SelectStatement` or `Table`